### PR TITLE
Refactor/Add tree service to the config framework

### PR DESCRIPTION
**Pull request by carpawell**

### CODE REVIEW

File: config/example/node.yaml [hunk 1]
```diff
@@ -60,10 +60,10 @@ grpc:
...
 
 tree:
-  enabled: false
-  cache_size: 10
-  replication_worker_count: 64
-  replication_channel_capacity: 64
+  enabled: true
+  cache_size: 15
+  replication_worker_count: 32
+  replication_channel_capacity: 32
```
Comment (fyrchik) [inline config/example/node.yaml:66]: The support is experimental, I don't think we should `enabled: true` in the example.

File: cmd/neofs-node/config/tree/config.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	// DefaultCacheSize is a default cache size of the tree service.
+	DefaultCacheSize = 10
+
+	// DefaultReplicationChannelCapacity is a default tree service
+	// replication channel capacity.
+	DefaultReplicationChannelCapacity = 64
+
+	// DefaultReplicationWorkerCount is a default tree service
+	// replication worker count.
+	DefaultReplicationWorkerCount = 64
```
Comment (fyrchik) [inline cmd/neofs-node/config/tree/config.go:17]: Let's not introduce defaults also here? We already have some in the service itself.

Comment (carpawell) [reply cmd/neofs-node/config/tree/config.go:17]: ok, made them return 0

File: config/example/node.yaml (hunk 1 above)
Comment (carpawell) [reply config/example/node.yaml:66]: yes, the only reason why i did so is unit tests that distinguish default and non-default values

is it ok to keep the default values in examples?

Comment (carpawell) [reply config/example/node.yaml:66]: i meant `enabled: false`

we can not test that if our example files contain default values

Comment (fyrchik) [reply config/example/node.yaml:66]: Ok, I think we can make it `true`, we have `testnet`/`mainnet` examples anyway.

**Review by fyrchik — APPROVED** [review]

**Pull request merged**